src/reactA11yProptypesRule.ts

```diff
@@ -0,0 +1,197 @@
+/**
+ * Enforce ARIA state and property values are valid.
+ */
+
+import * as ts from 'typescript';
+import * as Lint from 'tslint/lib/lint';
+
+import { ExtendedMetadata } from './utils/ExtendedMetadata';
+import { getPropName, getStringLiteral } from './utils/JsxAttribute';
+import { IAria } from './utils/attributes/IAria';
+import {
+    isStringLiteral,
+    isNumericLiteral,
+    isJsxExpression,
+    isFalseKeyword,
+    isTrueKeyword,
+    isNullKeyword
+} from './utils/TypeGuard';
+
+// tslint:disable-next-line:no-require-imports no-var-requires
+const aria: { [attributeName: string]: IAria } = require('./utils/attributes/ariaSchema.json');
+
+export function getFailureString(propName: string, expectedType: string, permittedValues: string[]): string {
+    switch (expectedType) {
+        case 'tristate':
+            return `The value for ${propName} must be a boolean or the string 'mixed'.`;
+        case 'token':
+            return `The value for ${propName} must be a single token from the following: ${permittedValues}.`;
+        case 'tokenlist':
+            return `The value for ${propName} must be a list of one or more tokens from the following: ${permittedValues}.`;
+        case 'boolean':
+        case 'string':
+        case 'integer':
+        case 'number':
+        default: // tslint:disable-line:no-switch-case-fall-through
+            return `The value for ${propName} must be a ${expectedType}.`;
+    }
+}
+
+export class Rule extends Lint.Rules.AbstractRule {
+    public static metadata: ExtendedMetadata = {
+        ruleName: 'react-a11y-proptypes',
+        type: 'maintainability',
+        description: 'Enforce ARIA state and property values are valid.',
+        options: null,
+        issueClass: 'Non-SDL',
+        issueType: 'Warning',
+        severity: 'Important',
+        level: 'Opportunity for Excellence',
+        group: 'Accessibility'
+    };
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        return sourceFile.languageVariant === ts.LanguageVariant.JSX
+            ? this.applyWithWalker(new ReactA11yProptypesWalker(sourceFile, this.getOptions()))
+            : [];
+    }
+}
+
+class ReactA11yProptypesWalker extends Lint.RuleWalker {
+    public visitJsxAttribute(node: ts.JsxAttribute): void {
+        const propName: string = getPropName(node).toLowerCase();
+
+        // if there is not a aria-* attribute, don't check it.
+        if (!aria[propName]) {
+            return;
+        }
+
+        const allowUndefined: boolean = aria[propName].allowUndefined || false;
```

Comment:
Although the logic is the same, I'd prefer using
aria[propName].allowUndefined !== undefined
  ? aria[propName].allowUndefined
  : false